telemetry/spans.go

```diff
@@ -73,11 +76,44 @@ func (s *Span) writeJSON(buf *bytes.Buffer) {
 	}
 
 	internal.AddAttributes(&ww, s.Attributes)
-
 	buf.WriteByte('}')
+
+	if len(s.Events) > 0 {
+		w.AddKey("events")
+		buf.WriteByte('[')
+		for i, e := range s.Events {
+			if i > 0 {
+				buf.WriteByte(',')
+			}
+			buf.WriteByte('{')
+			aw := internal.JSONFieldsWriter{Buf: buf}
+			aw.StringField("name", e.Name)
+			aw.IntField("timestamp", e.Timestamp.UnixNano()/(1000*1000))
+			aw.AddKey("attributes")
+			buf.WriteByte('{')
+			aw.NoComma()
```

Comment:
Since `aw` is a new JSONFieldsWriter, separate from `w`, doesn't it have a fresh, `false` copy of w.needsComma already? I'm not seeing why we need to set it to false again.

Reply:
The call to AddAttributes is going to prepend a comma, because it doesn't expect the attributes to be the first things in the json object. 

So, in this case, the json ends up looking something like:
```json
  "attributes": {,"key":"value"}
```

This works in other uses, because attributes aren't the first thing in the object (see lines 65-78 above). Though now that I'm re-reading those lines, it would be possible to get a leading comma if the span has no Name, ParentId, Duration, or ServiceName. This would be interesting to test and fix (though I'd prefer to fix that outside of this PR).

Reply:
👍